### PR TITLE
ao_audiotrack: make audiotrack jni multi-instance and multi-thread safe

### PR DESCRIPTION
**Pull request by davincerepo**
The detailed issue is here:
https://github.com/mpv-player/mpv/issues/15212

problem: Since The AudioTrack is not an mpv instance level but a global object, it cannot support multiple mpv instances at the same time. For example, if you create two instances and then destroy one of them, the other instance may crash.

Add jni usage count to fix this.

### CODE REVIEW

File: audio/out/ao_audiotrack.c [hunk 1]
```diff
@@ -524,26 +527,37 @@ static int AudioTrack_write(struct ao *ao, int len)
...
+            }
         }
     }
+    jni_static_use_count++;
+    mp_mutex_unlock(&jni_static_lock);
     return 0;
 
 error:
     uninit_jni(ao);
+    mp_mutex_unlock(&jni_static_lock);
```
Comment (sfan5) [inline audio/out/ao_audiotrack.c:560]: deadlock here because `uninit_jni` also tries to lock

File: audio/out/ao_audiotrack.c [hunk 2]
```diff
@@ -524,26 +527,37 @@ static int AudioTrack_write(struct ao *ao, int len)
...
 static void uninit_jni(struct ao *ao)
 {
-    JNIEnv *env = MP_JNI_GET_ENV(ao);
-    for (int i = 0; i < MP_ARRAY_SIZE(jclass_list); i++) {
-        mp_jni_reset_jfields(env, jclass_list[i].fields,
-                             jclass_list[i].mapping, 1, ao->log);
+    mp_mutex_lock(&jni_static_lock);
+    jni_static_use_count--;
+    if (jni_static_use_count == 0) {
+        JNIEnv *env = MP_JNI_GET_ENV(ao);
```
Comment (sfan5) [inline audio/out/ao_audiotrack.c:533]: a further safety check could be added here to ensure that two different mpv instances use the same `JNIEnv`.

**Review by sfan5 — COMMENTED** [review]
thanks for fixing this, some comments:

File: audio/out/ao_audiotrack.c (hunk 1 above)
Comment (davincerepo) [reply audio/out/ao_audiotrack.c:563]: Yes, I'll fix this.

File: audio/out/ao_audiotrack.c (hunk 2 above)
Comment (davincerepo) [reply audio/out/ao_audiotrack.c:533]: This should be fine. `Jclass` is a global object, and `JNIEnv` is a thread local object. As long as the `JNIEnv` use the same `JavaVM` behind, the `Jclass` object created through `NewGlobalRef` should be shared. And we can only have one `JavaVM` per android process.

Please check this document [JavaVM and JNIEnv](https://developer.android.com/training/articles/perf-jni#javavm-and-jnienv) 
> In theory you can have multiple JavaVMs per process, but Android only allows one.
> The JNIEnv is used for thread-local storage. For this reason, you cannot share a JNIEnv between threads.

Comment (sfan5) [reply audio/out/ao_audiotrack.c:533]: I'm aware it's a theoretical concern. We can leave this check out.

**Review by sfan5 — APPROVED** [review]
works fine, thanks.

**Pull request merged**